dds/DCPS/transport/rtps_udp/MetaSubmessage.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include "MetaSubmessage.h"
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+namespace DCPS {
+
+bool MetaSubmessage::operator==(const MetaSubmessage& other) const
+{
+  ACE_Message_Block left_block(1024);
```

Comment:
Instead of a fixed length with a hard-coded constant, this could get the serialized size of the object.
Or if that's not important, at least check if serialization succeeded.